nvflare/fuel/hci/client/api.py

```diff
@@ -164,9 +170,56 @@ def _load_client_cmds(self, cmd_modules):
     def register_command(self, cmd_entry):
         self.all_cmds.append(cmd_entry.name)
 
+    def start_session_monitor(self, session_callback, interval=5):
+        if self.sess_monitor_thread and self.sess_monitor_thread.is_alive():
+            self.close_session_monitor()
+        self.sess_monitor_thread = threading.Thread(
+            target=self._check_session, args=(session_callback, interval), daemon=True
+        )
+        self.sess_monitor_active = True
+        self.sess_monitor_thread.start()
+
+    def close_session_monitor(self):
+        self.sess_monitor_active = False
+        if self.sess_monitor_thread:
+            self.sess_monitor_thread.join()
+
+    def _check_session(self, session_callback, interval):
+        error_msg = ""
+        connection_error_counter = 0
+        while True:
+            time.sleep(interval)
+
+            if not self.sess_monitor_active:
+                return
```

Comment:
What if user call `start_session_monitor` and then call `close_session_monitor` right away?

In that case the `self.api_active` remains True and `session_callback` is not called?

Reply:
If they call these sequentially in a single thread it will be ok, but I agree, if they call them in separate threads there could be issues. I can add a lock.

Reply:
got it, I think we can assume single thread for this api.
In this case, do we need this if statement?

Reply:
(resolved offline) Assuming api is single threaded in this pr, but may change to support multi-threads in the future.